java/rocksjni/logger_from_options.cc

```diff
@@ -0,0 +1,95 @@
+//  Copyright (c) Meta Platforms, Inc. and affiliates.
+//
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#include <jni.h>
+
+#include <memory>
+
+#include "include/org_rocksdb_util_LoggerFromOptions.h"
+#include "rocksjni/cplusplus_to_java_convert.h"
+#include "rocksjni/portal.h"
+#include "util/stderr_logger.h"
+
+/*
+ * Create an "independent" logger, such as might be supplied to a readonly DB
+ * (e.g. in a readonly filesystem)
+ *
+ * Class:     org_rocksdb_util_LoggerFromOptions
+ * Method:    newLoggerFromOptions
+ * Signature: (J)J
+ */
+JNIEXPORT jlong JNICALL
+Java_org_rocksdb_util_LoggerFromOptions_createLoggerFromOptions(
+    JNIEnv* env, jclass, jstring jdb_name, jlong joptions_handle) {
+  auto* db_options =
+      reinterpret_cast<ROCKSDB_NAMESPACE::DBOptions*>(joptions_handle);
+
+  if (jdb_name == nullptr) {
+    ROCKSDB_NAMESPACE::RocksDBExceptionJni::ThrowNew(
+        env,
+        ROCKSDB_NAMESPACE::Status::InvalidArgument("Invalid (null) db name."));
+    return 0;
+  }
+  jboolean has_exception = JNI_FALSE;
+  auto db_name = ROCKSDB_NAMESPACE::JniUtil::copyStdString(
+      env, jdb_name, &has_exception);  // also releases jlog_prefix
+  if (has_exception == JNI_TRUE) {
+    return 0;
+  }
+
+  std::shared_ptr<ROCKSDB_NAMESPACE::Logger> logger;
+
+  ROCKSDB_NAMESPACE::Status s =
+      ROCKSDB_NAMESPACE::CreateLoggerFromOptions(db_name, *db_options, &logger);
+  if (!s.ok()) {
+    ROCKSDB_NAMESPACE::RocksDBExceptionJni::ThrowNew(env, s);
+    return 0;
+  }
+
+  return reinterpret_cast<jlong>(
```

Comment:
Corrent me if I'm wrong, but I thould we should use macro `GET_CPLUSPLUS_POINTER` to get C++ pointer. Or here it's not possible because of `std::move`